File: Homework6/Program/Presenter/Presenter.java
package Homework6.Program.Presenter;

import Homework6.Program.View.Menu;
import Homework6.Program.Model.Calculator;
import Homework6.Program.View.ConsoleView;

public class Presenter {
    public void run() {
        ConsoleView view = new ConsoleView();
        
        String[] mainMenuItems = new String[] {
                "Вычислить выражение",
                "Установить точность",
                "Выход" };
        Menu mainMenu = new Menu(view, mainMenuItems);
        
        String prompt;
        String choice;
        String expr;

        // test expression =
        // "15/(7-(1+1))*3-(2+(1+1))*15/(7-(200+1))*3-(2+(1+1))*(15/(7-(1+1))*3-(2+(1+1))+15/(7-(1+1))*3-(2+(1+1)))";
        // result is -30,072165

        Calculator calc = new Calculator();
        boolean stopMainMenu = false;
        
        do {
            mainMenu.show();
            choice = mainMenu.getUserChoice("Ваш выбор: ");
            switch (choice) {
                case "1":
                    prompt = "Введите выражение: ";
                    expr = view.getUserInput(prompt);
                    String result = String.format(
                            "\nРезультат: %s\n\n", calc.calculate(expr));
                    view.printMessage(result);
                    break;
                case "2":
                    prompt = "Текущая точность: " + calc.getAccuracy() + " знака.";
                    view.printMessage(prompt);
                    prompt = "Введите желаемую точность";
                    try {
                        int accuracy = Integer.parseInt(view.getUserInput(prompt));
                        if (!calc.setAccuracy(accuracy)) {
                            view.printMessage("Неверный ввод!");
                        } else {
                            view.printMessage("Установлена точность: " + calc.getAccuracy() + " знаков");
                        }
                    } catch (Exception e) {
                        System.out.println("Неверный ввод!");
                    }
                    break;
                case "3":
                    prompt = "До свидания!";
                    stopMainMenu = true;
                    view.printMessage(prompt);
                    break;
                default:
                    prompt = "Неверный ввод!";
                    view.printMessage(prompt);
                    break;
            }
        } while (!stopMainMenu);
    }
}

File: Homework7/Presenter.py
import sys
sys.path.insert(
    0, "c:\\Users\\Asus\\Documents\\Обучение GeekBrains\\repo\\OOP\\OOP_HW\\Homework7\\Model")
sys.path.insert(
    0, "c:\\Users\\Asus\\Documents\\Обучение GeekBrains\\repo\\OOP\\OOP_HW\\Homework7\\View")
import menu
import view
import Calc

# exp_str = '15/(7-(1+1))*3-(2+(1+1))*15/(7-(200+1))*3-(2+(1+1))*(15/(7-(1+1))*3-(2+(1+1))+15/(7-(1+1))*3-(2+(1+1)))'
# res = -30.072164948453608
calc = Calc.Calculator()
view = view.View()

main_menu_items = ["Вычислить выражение", "Задать точность", "Выход"]
my_menu = menu.Menu(main_menu_items)

while True:
    choice = my_menu.get_user_choice()
    if choice == '1':
        exp_str = view.get_user_input('Введите выражение: ')
        view.print_message(f'Результат: {calc.calculate(exp_str)}')
    elif choice == '2':
        pass
    elif choice == '3':
        view.print_message('До новых встреч!')
        break
